packages/typespec-test/test/translator/tspconfig.yaml

```diff
@@ -6,6 +6,7 @@ options:
     generateMetadata: true
     generateTest: true
     azureSdkForJs: false
+    compatibilityMode: true
```

Comment:
`compatabilityMode` is used outside of this area too, right? Do we have documentation somewhere on what exactly `compatabilityMode` does that I can look at? If not, I think a centralized doc which explains what changes when you enable the flag would be quite helpful.

Reply:
sure, i will add the doc to express that.